File: algorithms/0389.find_the_difference/find_the_difference.cpp
class Solution {
public:
    char findTheDifference(const string& s, const string& t) {
        char c = t.back();
        for (size_t i = 0; i < s.size(); ++i) {
            c ^= (s[i] ^ t[i]);
        }
        return c;
    }
};

File: algorithms/0389.find_the_difference/find_the_difference.rs
impl Solution {
    pub fn find_the_difference(s: String, t: String) -> char {
        let (s, t) = (s.as_bytes(), t.as_bytes());
        let mut ch = *t.last().unwrap();
        for i in 0..s.len() {
            ch ^= s[i] ^ t[i];
        }
        ch as char
    }
}